src/main/java/com/example/demo/services/impl/UserServiceImpl.java

```diff
@@ -52,19 +68,34 @@ public User getUserById(final long id) {
 
     @Override
     public void addUser(final UserCreation user) {
+        if (user.getAge() < 0) {
+            log.error(ageValidError);
+            throw new RuntimeException(ageValidError);
```

Comment:
For update and add you have the same age validation. Extract common code to the separate method. DRY - don't repeat yourself